cylc/flow/prerequisite.py

```diff
@@ -194,20 +204,25 @@ def _conditional_is_satisfied(self):
                 '"%s":\n%s' % (self.get_raw_conditional_expression(), err_msg))
         return res
 
-    def satisfy_me(self, all_task_outputs):
-        """Evaluate prerequisite against known outputs.
+    def satisfy_me(self, outputs: Iterable['Tokens']) -> 'Set[Tokens]':
+        """Attempt to satisfy me with given outputs.
 
-        Updates cache with the evaluation result.
+        Updates cache with the result.
+        Return outputs that match.
 
         """
-        relevant_messages = all_task_outputs & set(self.satisfied)
-        for message in relevant_messages:
-            self.satisfied[message] = self.DEP_STATE_SATISFIED
+        valid = set()
+        for output in outputs:
+            prereq = output.to_prereq_tuple()
+            if prereq not in self.satisfied:
```

Comment:
prereq is an output tuple (`(point, name, output)`) , self.satisfied contains message tuples: `(point, name, message)`, so this comparison won't work.

Reply:
(This has been resolved)